### PR TITLE
Fix imx477.h mode numbering to match documentation, add tools for HQ modes

### PR DESCRIPTION
**Pull request by Hermann-SW**
The numbering of modes in imx477.h did not match documentation.
2028x1520 and 2028x1080 were switched, 1012x716 was mode3 and not mode4.

```
🍓 1012x760 4000 | grep -v "^[> ]"
removing /dev/shm/out.*.raw
capturing frames for 4000ms with 195fps requested
620 frames were captured
majority framerate 197fps
frame delta time[us] distribution
after skip frame indices (middle column)
21% frame skips
average framerate 155fps
🍓
```

### CODE REVIEW

File: tools/gifenc.sh [hunk 1]
```diff
@@ -6,7 +6,8 @@
 palette="/tmp/palette.png"
 
 #filters="fps=15,scale=320:-1:flags=lanczos"
-filters="fps=25,scale=640:-1:flags=lanczos"
+#filters="fps=25,scale=640:-1:flags=lanczos"
+filters="fps=25,scale=1012:-1:flags=lanczos"
```
Comment (6by9) [inline tools/gifenc.sh:10]: Unrelated change, or at least undocumented in commit text.

Comment (Hermann-SW) [reply tools/gifenc.sh:10]: For ov5647 and imx219 the modes match camera documentation.
For imx477 they did not.
Do you want to take changed mode numbers (0123->02103) as I did to match Raspberry camera documentation?
If so, how can I modify my changelist into 2 (mode renumbering / 4 HQ tools+gifenc.sh change) without deleting my fork, forking new and do changes based on your latest from fresh?
I just commited camera_i2c fix on my fork before reading your comment, and the commit contains camera_i2c i2c device change as well as changes for the 4 tools. In case you want the camera_i2c change, I will need to split that as well.

**Comment by 6by9** [comment]
Throwing extra stuff into a PR and not addressing the issues raised is not going to help this get merged.
I do not want to see your work in progress patches updating line_time_ns from 20000 to 28560 to 28571. ```git rebase -i``` will allow you to squash the commits together, and then you can force-push to your branch.

```git rebase -i``` will also allow you to edit or reorder existing patches. Take a copy of the files for one part of your patch to be split. Undo the changes (```git checkout <hash> <files>``` is the easiest way) and amend the one patch (```git commit --amend <files>```). Copy your backed up files back in, and create a new patch with them.

**Comment by Hermann-SW** [comment]
Thank you for the instructions, I will create a new pull request.

**Comment by 6by9** [comment]
You don't need to create a new pull request, you only needed to update the source branch that you'd created this PR from (https://github.com/Hermann-SW/raspiraw master branch).
